### ISSUE TITLE
[stage/02] Metadata enrichment — entities, topics, knowledge density, temporal sensitivity

### ISSUE THREAD

**Issue by Kabuswe** [Kabuswe/pipeline-finetune#3]
# [stage/02] Metadata enrichment — entities, topics, knowledge density, temporal sensitivity

## Purpose
Implement `stages/s02_enrich.py`. For each chunk produced by Stage 01, call a local Ollama model (configured in `config.enrichment.model`) to generate structured metadata. Output writes enrichment fields back to the document registry and marks documents `status='enriched'`. This metadata drives intelligent JSONL generation in Stage 03.

## File Structure
```
stages/
  s02_enrich.py             ← stage runner
enricher/
  __init__.py
  chunk_enricher.py         ← per-chunk enrichment logic
  batch_runner.py           ← parallel batch processing with rate limiting
  prompts.py                ← enrichment prompt templates
```

## Enrichment Schema
The `ChunkMetadata` output from the LLM (structured output via JSON mode):

```python
@dataclass
class ChunkMetadata:
    chunk_id: str
    entities: list[str]           # named entities: tools, concepts, people, organisations
    topic_labels: list[str]       # 2–5 short topic strings
    content_type: Literal["factual", "procedural", "conceptual", "opinion"]
    knowledge_density: float      # 0–1: extractable knowledge per token (0=low, 1=high)
    temporal_sensitivity: Literal["stable", "versioned", "ephemeral"]
    # stable    = evergreen content, safe to bake into weights
    # versioned = tied to a software version, include version in QA pairs
    # ephemeral = changelog/date-specific, skip for training unless calibration pairs
    questionability: float        # 0–1: likelihood this chunk generates good QA pairs
    estimated_qa_pairs: int       # derived: round(questionability * knowledge_density * 10)
    skip_training: bool           # True if knowledge_density < threshold OR ephemeral
```

## Enrichment Prompt (`enricher/prompts.py`)
The prompt must be domain-agnostic — no domain name injected. The model infers domain from content:

```
Analyse the following text chunk and return a JSON object with these fields:
- entities: list of named entities (tools, frameworks, concepts, people, orgs)
- topic_labels: 2–5 short topic strings
- content_type: one of factual | procedural | conceptual | opinion
- knowledge_density: float 0–1 (how much extractable knowledge per token)
- temporal_sensitivity: one of stable | versioned | ephemeral
- questionability: float 0–1 (how likely this generates good QA training pairs)

Text chunk:
---
{content}
---

Return only valid JSON, no explanation.
```

## Batch Runner
- Process chunks in batches of `config.enrichment.batchSize` (default: 20)
- Use `concurrent.futures.ThreadPoolExecutor` for parallel Ollama calls
- Respect Ollama rate limits: if a call fails with a timeout, retry with backoff (max 3 attempts)
- Progress bar via `tqdm`
- Skip chunks where `skip_training=True` early (no LLM call needed for ephemeral chunks below density threshold)

## Stage Runner Logic
```python
def run(config: PipelineConfig):
    docs = get_documents_by_status(config.domain, 'raw')
    for doc in docs:
        enriched_chunks = []
        for chunk in doc.chunks:
            if chunk_below_threshold(chunk, config.enrichment):
                metadata = ChunkMetadata(..., skip_training=True)
            else:
                metadata = enrich_chunk(chunk, config.enrichment.model)
            enriched_chunks.append(metadata)
        update_document_enrichment(doc.doc_id, enriched_chunks)
        mark_document_status(doc.doc_id, 'enriched')
```

## Skip Logic
A chunk is skipped (no LLM enrichment call, `skip_training=True`) if:
- Token count < 50 (too short to generate useful pairs)
- The chunk is a table of contents, header-only, or navigation element (detect via simple heuristics)
- `config.enrichment.skipBelowKnowledgeDensity` threshold is set and the pre-filter estimates density below it (use token entropy as a fast proxy before calling LLM)

## Acceptance Criteria
- All chunks from Stage 01 that aren't skipped receive a `ChunkMetadata` record
- `knowledge_density` and `questionability` values fall in [0, 1]
- `estimated_qa_pairs` is correctly computed as `round(questionability * knowledge_density * 10)`
- Chunks marked `skip_training=True` are excluded from the `estimated_qa_pairs` sum
- Re-running Stage 02 on already-enriched documents skips them (idempotent)
- Batch processing 100 chunks completes in < 5 minutes on a machine running Ollama with llama3.2:3b

## Dependencies
- Issue #1 (registry schema)
- Issue #2 (Stage 01 ingestion) — documents must exist with `status='raw'`

## Labels
`stage/02-enrichment` `priority/p0`
